packages/aws-cdk-lib/aws-stepfunctions/lib/util.ts

```diff
@@ -1,4 +1,37 @@
+import { IKey } from '../../aws-kms';
+import { Duration } from '../../core';
+
 export function noEmptyObject<A>(o: Record<string, A>): Record<string, A> | undefined {
   if (Object.keys(o).length === 0) { return undefined; }
   return o;
+}
+
+function isInValidKmsDataKeyReusePeriodSeconds(kmsDataKeyReusePeriodSeconds: Duration) {
+  return kmsDataKeyReusePeriodSeconds.toSeconds() < 60 || kmsDataKeyReusePeriodSeconds.toSeconds() > 900;
+}
+
+function validateEncryptionConfiguration(kmsKey: IKey | undefined, kmsDataKeyReusePeriodSeconds: Duration | undefined) {
+  if (!kmsKey && kmsDataKeyReusePeriodSeconds) {
+    throw new Error('You cannot set kmsDataKeyReusePeriodSeconds without providing a value for kmsKey');
+  }
```

Comment:
This check won't be needed anymore after combining the `kmsKey` and `kmsDataKeyReusePeriodSeconds` properties into a single `KmsConfig` class.

Reply:
Makes sense, will remove.